### PR TITLE
B2B Logic: Dynamic Contact-Header + Trigger Scenario from Script

### PR DESCRIPTION
**Pull request by carstenbock**
**Summary**
This PR adds the following functionality:
- Allow pseudo-variables in the contact header
- Trigger a scenario from the script (e.g. for out-of-dialog REFERs)

**Details**
1) The Contact-Header parameter has some limitations, as it becomes challenging when having different aliases defined for different sockets. With this PR, you can use all kinds of Pseudo-Variables for the Contact header, e.g. the Contact header could even be defined in the script.
2) Added functionality to trigger a scenario from Script. This may be used to for example handle an out of dialog REFER or if you want to process multiple in-dialog REFERs without touching the initial session.

**Compatibility**
No impact on existing configurations, the default behavior stays the same.

### CODE REVIEW

File: modules/b2b_logic/logic.c [hunk 1]
```diff
@@ -4106,6 +4134,360 @@ int b2bl_bridge(str* key, str* new_dst, str *new_proxy, str* new_from_dname,
...
+	tuple = b2bl_insert_new(msg, hash_index, &init_params,
+		NULL, NULL, -1, &b2bl_key, INSERTDB_FLAG, TUPLE_NO_REPL);
+	if(tuple== NULL)
+	{
+		LM_ERR("Failed to insert new scenario instance record\n");
+		rc = -1;
+		goto end;
+	}
+	tuple->lifetime = 60 + get_ticks();
+	LM_ERR("Key: %.*s (%p)\n", b2bl_key->len, b2bl_key->s, tuple);
```
Comment (rvlad-patrascu) [inline modules/b2b_logic/logic.c:4365]: This should be a LM_DBG log.

File: modules/b2b_logic/logic.c [hunk 2]
```diff
@@ -4106,6 +4134,360 @@ int b2bl_bridge(str* key, str* new_dst, str *new_proxy, str* new_from_dname,
...
+			goto end;
+		}
+		if (s->s[0] == 'n') {
+			tuple->bridge_flags = B2BL_BR_FLAG_NOTIFY | B2BL_BR_FLAG_DONT_DELETE_BRIDGE_INITIATOR;
+			s = param_list->next ? &param_list->next->s : NULL;
+			if (s && s->s && s->len) {
+				remote_tuple = s;
+				s = param_list->next->next ? &param_list->next->next->s : NULL;
+				if (s && s->s && s->len) {
+					if (s->s[0] == 1) remote_tuple_party = 1;
```
Comment (rvlad-patrascu) [inline modules/b2b_logic/logic.c:4387]: It seems that this param identifies the old entity that receives the NOTIFY, given as an index in the bridged entities array, so the comparison should probably be `if (s->s[0] == '1')`. Can you also clarify a bit in the docs that this is an extra parameter besides the b2b_logic key?

**Review by rvlad-patrascu — COMMENTED** [review]
Besides the small fixes suggested, the PR looks ok and ready to be merged.

File: modules/b2b_logic/logic.c (hunk 1 above)
Comment (carstenbock) [reply modules/b2b_logic/logic.c]: Found a few more places, where it should have been LM_DBG => resolved!

File: modules/b2b_logic/logic.c (hunk 2 above)
Comment (carstenbock) [reply modules/b2b_logic/logic.c]: Improved the comparison and improved the docs. Thank you for noticing!

**Review by rvlad-patrascu — APPROVED** [review]

**Pull request merged**